File: webui/src/test/scala/com/taskadapter/webui/uiapi/ConfigFolderTestConfigurer.scala
package com.taskadapter.webui.uiapi

import java.io.File
import java.nio.file.{Files, Paths}

import com.taskadapter.web.uiapi.SetupId

object ConfigFolderTestConfigurer {

  val jiraSetupId = SetupId("Atlassian_JIRA_1.json")
  val jiraCloudSetupId = SetupId("Atlassian_Jira_cloud.json")
  val redmineSetupId = SetupId("Redmine_1.json")

  /**
    * Configures test config folder with JIRA/Redmine credentials.
    *
    * @return folder with user configs
    */
  def configure(rootFolder: File): Unit = {
    List(jiraSetupId.id,
      jiraCloudSetupId.id,
      redmineSetupId.id,
      SetupId("Microsoft_Project_1.json").id,
      SetupId("GitHub1.json").id,
      SetupId("Mantis1.json").id
    )
      .foreach { resourceName =>
        val adminFolder = new File(rootFolder, "admin")
        adminFolder.mkdirs()
        Files.copy(Paths.get(getClass.getClassLoader.getResource(resourceName).getPath),
          new File(adminFolder, resourceName).toPath)
      }
  }
}

File: webui/src/test/java/com/taskadapter/web/uiapi/ConfigFolderTestConfigurer.java
package com.taskadapter.web.uiapi;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;

public class ConfigFolderTestConfigurer {
    public static SetupId jiraSetupId = new SetupId("Atlassian_JIRA_1.json");
    public static SetupId jiraCloudSetupId = new SetupId("Atlassian_Jira_cloud.json");
    public static SetupId redmineSetupId = new SetupId("Redmine_1.json");

    /**
     * Configures test config folder with JIRA/Redmine credentials.
     *
     * @return folder with user configs
     */
    public static void configure(File rootFolder) {
        List.of(jiraSetupId.getId(),
                jiraCloudSetupId.getId(),
                redmineSetupId.getId(),
                new SetupId("Microsoft_Project_1.json").getId(),
                new SetupId("GitHub1.json").getId(),
                new SetupId("Mantis1.json").getId()
        ).forEach(resourceName -> {
            var adminFolder = new File(rootFolder, "admin");
            adminFolder.mkdirs();
            try {
                Files.copy(Paths.get(ConfigFolderTestConfigurer.class.getClassLoader().getResource(resourceName).getPath()),
                        new File(adminFolder, resourceName).toPath());
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        });
    }
}
